db/prompts.go

```diff
@@ -158,11 +148,56 @@ var AllowedColumns = map[string]bool{
 }
 
 var usesField = `array(SELECT prompts_uses.created_at FROM prompts_uses WHERE prompts_uses.prompt_id = prompts.id ) as uses`
-var minField = `prompts.id, prompts.name, prompts.description, prompts.use, prompts.tags, prompts.public, prompts.user_id`
-var maxField = `prompts.id, prompts.name, prompts.description, prompts.prompt, prompts.created_at, prompts.updated_at, prompts.tags, prompts.public, prompts.user_id`
+var likesField = `(SELECT COUNT(*) FROM prompts_likes WHERE prompts_likes.prompt_id = prompts.id) as likes`
+
+var minField = `prompts.id, prompts.name, prompts.description, prompts.use, prompts.tags, prompts.public, prompts.user_id, prompts.slug`
+var maxField = `prompts.id, prompts.name, prompts.description, prompts.prompt, prompts.created_at, prompts.updated_at, prompts.tags, prompts.public, prompts.user_id, prompts.slug`
+
+func GetPromptByIdOrSlug(id string) (*Prompt, error) {
+	fieldMappings := map[string]string{
```

Comment:
The fieldMappings should be in the PrepareQueryByStringIdentified function. Here it reads as if the function should be trusted but sending a user generated fieldMappings could cause an SQL injection. Which is confusing imo